### ISSUE TITLE
[React Tooltip] carry over a11y fixes from Vanilla

### ISSUE THREAD

**Issue by elizabethsjudd** [carbon-design-system/carbon#3812]
Multiple updates have been made to the interactive tooltip for a11y.  These changes should be carried over to React

https://github.com/carbon-design-system/carbon/pull/3148
https://github.com/carbon-design-system/carbon/pull/3476

**Comment by emyarod** [comment]
this should be closed by https://github.com/carbon-design-system/carbon/pull/3840 and https://github.com/carbon-design-system/carbon/pull/3842

**Comment by emyarod** [comment]
actually might need to take another look as this is related to interactive tooltip
